PUPPETVERSIONDescription: Upstream changes introduced in version 2.6.1-0ubuntu2
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 puppet (2.6.1-0ubuntu2) maverick; urgency=low
 .
   * debian/puppetmaster-passenger.postinst: Use cacrl instead of hostcrl to
     set the location of the CRL in apache2 configuration. Fix apache2
     configuration on upgrade as well (LP: #641001).
 .
 The person named in the Author field signed this changelog entry.
Author: Mathias Gug <mathiaz@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/641001

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- puppet-2.6.1.orig/test/lib/puppettest/fakes.rb
+++ puppet-2.6.1/test/lib/puppettest/fakes.rb
@@ -1,4 +1,4 @@
-require File.expand_path(File.join(File.dirname(__FILE__), '../../../lib/puppet/util'))
+require '/usr/lib/ruby/1.8/puppet/util'

 module PuppetTest
   # A baseclass for the faketypes.
--- puppet-2.6.1.orig/lib/puppet/provider/service/init.rb
+++ puppet-2.6.1/lib/puppet/provider/service/init.rb
@@ -134,7 +134,15 @@ Puppet::Type.type(:service).provide :ini
   # we just return that; otherwise, we return false, which causes it to
   # fallback to other mechanisms.
   def statuscmd
-    (@resource[:hasstatus] == :true) && [initscript, :status]
+      if @resource[:hasstatus] == :true then
+          # Workaround the fact that initctl status command doesn't return
+          # proper exit codes. Can be removed once LP: #552786 is fixed.
+          if File.symlink?(initscript) && File.readlink(initscript) == "/lib/init/upstart-job" then
+              ['sh', '-c', "LANG=C invoke-rc.d #{File::basename(initscript)} status | grep -q '^#{File::basename(initscript)}.*running'" ]
+          else
+              [initscript, :status ]
+          end
+      end
   end

 end
--- puppet-2.6.1.orig/ext/rack/files/apache2.conf
+++ puppet-2.6.1/ext/rack/files/apache2.conf
@@ -1,33 +1,4 @@
-# performance settings, from http://reductivelabs.com/trac/puppet/wiki/UsingPassenger
-
-# Set to 5 min (300 seconds) or less. The shorting this option allows for
-# puppetmasterd to get refreshed at some interval. This option is also
-# somewhat dependent upon the amount of puppetd nodes connecting and at what
-# interval.
-PassengerPoolIdleTime 300
-
-# to 15% more instances than what's needed. This will allow idle
-# puppetmasterd to get recycled. The net effect is less memory will be used,
-# not more.
-PassengerMaxPoolSize 15
-
-# Since communication with the puppetmaster from puppetd is a long process
-# (more than 20 seconds in most cases) and will allow for processes to get
-# recycled better
-PassengerUseGlobalQueue on
-
-# The additional Passenger features for apache compatibility are not needed
-# with Puppet.
-PassengerHighPerformance on
-
-# Whether Passenger should automatically detect whether a virtual host’s
-# document root is a Rack application. The default is on. Because
-# RackBaseURI is set, this does not need to be on
-RackAutoDetect Off
-
-# Whether Phusion Passenger should automatically detect whether a virtual
-# host’s document root is a Ruby on Rails application. The default is on.
-RailsAutoDetect Off
+# Based on http://projects.puppetlabs.com/projects/1/wiki/Using_Passenger

 Listen 8140

@@ -43,12 +14,24 @@ Listen 8140
         # If Apache complains about invalid signatures on the CRL, you can try disabling
         # CRL checking by commenting the next line, but this is not recommended.
         SSLCARevocationFile     /var/lib/puppet/ssl/ca/ca_crl.pem
+        # Set to require if this puppetmaster doesn't issue certificates
+        # to puppet clients.
+        # NB: this requires SSLCACertificateFile to include the CA cert
+        #     issuing puppet client certificate.
         SSLVerifyClient optional
         SSLVerifyDepth  1
         SSLOptions +StdEnvVars

-        DocumentRoot /usr/share/puppet/rack/puppetmasterd/public
+        # Passenger options that can be set in a virtual host
+        # configuration block.
+        PassengerHighPerformance on
+        PassengerStatThrottleRate 120
+        PassengerUseGlobalQueue on
+        RackAutoDetect Off
+        RailsAutoDetect Off
         RackBaseURI /
+
+        DocumentRoot /usr/share/puppet/rack/puppetmasterd/public
         <Directory /usr/share/puppet/rack/puppetmasterd/>
                 Options None
                 AllowOverride None
